### DISCUSSION TITLE
PyTorch Geometric SAGEConv - Expected scalar type Long, but found Float?

### DISCUSSION THREAD

**Discussion by BrandenKeck** [pyg-team/pytorch_geometric#9135] [General]
I am having tremendous trouble with my torch datatypes in the torch_geometric modules.  It seems that no matter what I adjust in my code, SAGEConv( ) is expecting a Long datatype and is receiving a Float datatype in one of its linear layers.   I also opened this issue to Stack Overflow here: [PyTorch Geometric SAGEConv - Expected scalar type Long, but found Float?](https://stackoverflow.com/questions/78255992/pytorch-geometric-sageconv-expected-scalar-type-long-but-found-float)

I have trying to troubleshoot using the most basic example I could come up with:

```python
import torch  
from torch_geometric.nn import SAGEConv  
sconv = SAGEConv((-1, -1), 64)  
x = torch.tensor([[1,0],[2,4],[5,7]])  
ei = torch.tensor([[1, 1],[0,2]])  
sconv(x, ei)  
```

Results in:
```
RuntimeError: expected m1 and m2 to have the same dtype, but got: __int64 != float
```

If I print my types, I get:
```python
x.type()
'torch.LongTensor'
ei.type()
'torch.LongTensor'
```

Any help with this is extremely appreciated.

**Comment by rusty1s** [comment]
`x` should be a floating point tensor:
```python
x = torch.tensor([[1,0],[2,4],[5,7]], dtype=torch.float)
